mcs/class/System/System.Diagnostics/PerformanceCounter.cs

```diff
@@ -138,6 +138,11 @@ public PerformanceCounter ()
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
 		static extern void FreeData (IntPtr impl);
 
+		static bool ValidMachine (string machine)
+		{ // no support for counters on other machines
+			return machine.Length == 1 && machine[0] == '.';
```

Comment:
For my information: Why this construct instead of `return machine == ".";`

Reply:
I'm not sure if that is a function call, and uses locale, and considers case sensitivity, and other complicated stuff. Perhaps it is identical.

Reply:
Perhaps your way is better.

Reply:
- The name should start with `IsValid....`
- `return machine == ".";` is ordinal comparison so it does what you wrote in expanded form

Reply:
Does it check length before looking at the characters?

Reply:
yes it does: https://github.com/mono/mono/blob/146346977cea8aaed2de8ab6a84c2a58b30aea24/mcs/class/referencesource/mscorlib/system/string.cs#L528-L542